eclair-core/src/main/scala/fr/acinq/eclair/UInt64.scala

```diff
@@ -16,40 +16,43 @@
 
 package fr.acinq.eclair
 
-import java.math.BigInteger
-
+import com.google.common.primitives.UnsignedLongs
 import scodec.bits.ByteVector
+import scodec.bits.HexStringSyntax
 
-case class UInt64(private val underlying: BigInt) extends Ordered[UInt64] {
+case class UInt64(private val underlying: Long) extends Ordered[UInt64] {
 
-  require(underlying >= 0, s"uint64 must be positive (actual=$underlying)")
-  require(underlying <= UInt64.MaxValueBigInt, s"uint64 must be < 2^64 -1 (actual=$underlying)")
+  override def compare(o: UInt64): Int = UnsignedLongs.compare(underlying, o.underlying)
+  private def compare(other: MilliSatoshi): Int = other.toLong match {
+    case l if l < 0 => 1                    // if @param 'other' is negative then is always smaller than 'this'
+    case _ => compare(UInt64(other.toLong)) // we must do an unsigned comparison here because the uint64 can exceed the capacity of MilliSatoshi class
+  }
 
-  override def compare(o: UInt64): Int = underlying.compare(o.underlying)
+  def <(other: MilliSatoshi): Boolean = compare(other) < 0
+  def >(other: MilliSatoshi): Boolean = compare(other) > 0
+  def <=(other: MilliSatoshi): Boolean = compare(other) <= 0
+  def >=(other: MilliSatoshi): Boolean = compare(other) >= 0
 
-  def toByteVector: ByteVector = ByteVector.view(underlying.toByteArray.takeRight(8))
+  def toByteVector: ByteVector = underlying match {
+    case 0 => hex"00"
+    case _ => ByteVector.fromLong(underlying).dropWhile(_ == 0)
+  }
 
-  def toBigInt: BigInt = underlying
+  def toBigInt: BigInt = BigInt(signum = 1, toByteVector.toArray)
```

Comment:
Feels a bit weird to go through `toByteVector` for converting to `BigInt`. Stackoverflow suggests:

```suggestion
  def toBigInt: BigInt = (BigInt(underlying >>> 1) << 1) + (underlying & 1)
```

source: https://stackoverflow.com/questions/21212993/unsigned-variables-in-scala

Tests pass with this change.

Reply:
The change have been implemented in 8c60d6a, while a bit less readable i think this solution is elegant and more efficient than going through the byte vector conversion.